etna/datasets/tsdataset.py

```diff
@@ -980,6 +980,56 @@ def has_hierarchy(self) -> bool:
         """Check whether dataset has hierarchical structure."""
         return self.hierarchical_structure is not None
 
+    def get_level_dataset(self, target_level: str) -> "TSDataset":
+        """Generate new TSDataset on target level.
+
+        Parameters
+        ----------
+        target_level:
+            target level name
+
+        Returns
+        -------
+        TSDataset
+            generated dataset
+        """
+        if self.hierarchical_structure is None or self.current_df_level is None:
+            raise ValueError("Method could be applied only to instances with a hierarchy!")
+
+        current_level_segments = self.hierarchical_structure.get_level_segments(level_name=self.current_df_level)
+        target_level_segments = self.hierarchical_structure.get_level_segments(level_name=target_level)
+
+        current_level_index = self.hierarchical_structure.get_level_depth(self.current_df_level)
+        target_level_index = self.hierarchical_structure.get_level_depth(target_level)
+
+        if target_level_index > current_level_index:
+            raise ValueError("Target level should be higher in the hierarchy than the current level of dataframe!")
+
+        if target_level_index < current_level_index:
+            summing_matrix = self.hierarchical_structure.get_summing_matrix(
+                target_level=target_level, source_level=self.current_df_level
+            )
+
+            source_level_index = pd.IndexSlice[current_level_segments, "target"]
+            source_level_data = self.df.loc[:, source_level_index].values
```

Comment:
Here it is better to use the implemented `__getitem__`of the dataset as it incapsulates what you have done.
```
 source_level_data = self[:, current_level_segments, "target"].values
```